### PR TITLE
Adding support for multiple worker node groups

### PR DESCRIPTION
**Pull request by maxdrib**
*Issue #, if available:*

*Description of changes:*
This PR adds basic functionality for multiple worker node groups in the cloudstack provider by aligning with the vsphere provider in the majority of cases

*Testing (if applicable):*
Unit tests pass. Successfully created cluster locally with 2 worker node groups. 
Screen Shot 2022-04-14 at 11 50 31 AM

Local E2E test passing, and will be added in next PR:

```
2022-04-18T11:45:23.594-0400	V3	Cleaning up long running container	{"name": "eksa_1650296193019641000"}
2022-04-18T11:45:23.990-0400	V3	e2e	Cleaning up long running container	{"name": "eksa_1650295285988707000"}
--- PASS: TestCloudStackKubernetes121RedhatAndAndRemoveWorkerNodeGroups (1438.27s)
PASS
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jaxesn** [comment]
@taneyland may be interested in reviewing

File: pkg/providers/cloudstack/cloudstack.go [hunk 1]
```diff
@@ -1057,29 +1051,43 @@ func (p *cloudstackProvider) DatacenterConfig(_ *cluster.Spec) providers.Datacen
...
+			configs[workerMachineName] = p.machineConfigs[workerMachineName]
+			if p.clusterConfig.IsManaged() {
+				p.machineConfigs[workerMachineName].SetManagement(p.clusterConfig.ManagedBy())
+			}
+		}
+	}
+	return configsMapToSlice(configs)
+}
+
+func configsMapToSlice(c map[string]providers.MachineConfig) []providers.MachineConfig {
```
Comment (vivek-koppuru) [inline pkg/providers/cloudstack/cloudstack.go:1085]: This is the same exact code right? Looks like it can be reused?

**Review by vivek-koppuru — APPROVED** [review]

File: pkg/providers/cloudstack/cloudstack.go (hunk 1 above)
Comment (maxdrib) [reply pkg/providers/cloudstack/cloudstack.go:1085]: Yes. However, I'm not sure we ever decided what to do with reusable code. Is `common` still the right package to be putting these functions? Or should we create some new package specifically for, say common MachineConfig operations?

**Review by vivek-koppuru — APPROVED** [review]

**Pull request merged**